tools/test-helpers/helpers.js

```diff
@@ -0,0 +1,35 @@
+window.D2L = window.D2L || {};
+window.D2L.TestHelpers = window.D2L.TestHelpers || {};
+
+export async function focus(selector, options) {
+	const elemToFocus = queryShadowSelector(selector);
+	if (!options || !options.waitFor) {
+		elemToFocus.focus();
+		return;
+	}
+	return new Promise((resolve) => {
+		const elemToWait = queryShadowSelector(options.waitFor.selector);
+		elemToWait.addEventListener(options.waitFor.eventName, resolve, { once: true });
+		elemToFocus.focus();
+	});
+}
+window.D2L.TestHelpers.focus = focus;
+
+export function queryShadowSelector(selectors, elem) {
```

Comment:
This helper accepts an string or array of selectors, and optionally a starting element.  

If a string selector is provided, it will simply use the native `querySelector`.  

If an array selector is provided, it will call `querySelector` as normal for the first selector in the array.  For subsequent selectors in the array it will call `querySelector` on `shadowRoot` to "drill" in.  If any fail, it will return `null`, as would the standard `querySelector`.